cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -286,4 +394,95 @@ public void finished(Throwable failureCause) {
       }
     }
   }
+
+  /**
+   * Aggregate the sizes of all stream writers. A notification is published if the aggregated
+   * size is higher than a threshold.
+   */
+  private final class Aggregator extends AbstractScheduledService {
```

Comment:
The ``Aggregator``classes in ``DistributedStreamService`` and ``LocalStreamService`` look very similar. I see that this one is an ``AbstractScheduledService`` and overrides its methods. Can the other methods be shared with the ``Aggregator`` class in ``LocalStreamService``? Some sort of a hierarchy?